### ISSUE TITLE
W0603 torch/fx/experimental/symbolic_shapes.py:1325] [1/18] RecursionError 

### ISSUE THREAD

**Issue by bhack** [pytorch/pytorch#127817]
### 🐛 Describe the bug

```python
in sympy.expand(Eq(Mod(256*evaluate_static_shape_1 + 512, 256*(evaluate_static_shape_0 + 1)*(evaluate_static_shape_2 + 1)), 0))
```

is there any action item for this?

### Versions

pytorch-nigthly

cc @ezyang @msaroufim @bdhirsh @anijain2305 @chauhang

**Comment by ezyang** [comment]
Is worth fixing. Curious if you have more context / a TORCH_TRACE.

**Comment by bhack** [comment]
Yes is that we have many rough edges/failures/warning on different `def` and also with different GPUs arch gencode(some things are failing on H100 but not on L4) that I am going `compile` disable and enable many different parts every nightly.
I need to find again this specific combination.

**Comment by ezyang** [comment]
for this one, the output of TORCH_LOGS=dynamic will be more useful, I think

**Comment by bhack** [comment]
Ok I've reproduced it but `TORCH_LOGS=dynamic` but the full it is really huge. Is there any  hint I could share?

**Comment by ezyang** [comment]
Just the logs right up to the error would be good to start, I think.

**Comment by bhack** [comment]
Here the log up the `RecusionError`
[dynamic.log](https://github.com/user-attachments/files/15598189/dynamic.log)

**Comment by ezyang** [comment]
huh is there stack trace there lol

**Comment by ezyang** [comment]
@IvanKobzarev This might be the same as the stack overflow thing you were seeing

**Comment by bhack** [comment]
After 9 months it is hard to re-ask for re-triage.
